src/commonMain/kotlin/exchange.dydx.abacus/calculator/TriggerOrdersInputCalculator.kt

```diff
@@ -99,68 +101,112 @@ internal class TriggerOrdersInputCalculator(val parser: ParserProtocol) {
         if (entryPrice != null) {
             val triggerPrice = parser.asDouble(parser.value(modified, "triggerPrice"))
             val usdcDiff = parser.asDouble(parser.value(modified, "usdcDiff"))
-            // val percentDiff = parser.asDouble(parser.value(modified, "percentDiff")) TODO: CT-694
+            val percentDiff = parser.asDouble(parser.value(modified, "percentDiff"))
 
             when (inputType) {
                 "stopLossOrder.price.triggerPrice" -> {
-                    modified.safeSet(
-                        "usdcDiff",
-                        if (triggerPrice != null) {
+                    if (triggerPrice != null) {
+                        modified.safeSet(
+                            "usdcDiff",
                             when (positionSide) {
                                 "long" -> size.times(entryPrice.minus(triggerPrice))
                                 "short" -> size.times(triggerPrice.minus(entryPrice))
                                 else -> null
-                            }
-                        } else {
-                            null
-                        },
-                    )
+                            },
+                        )
+                        modified.safeSet(
+                            "percentDiff",
+                            when (positionSide) {
+                                "long" -> leverage.times(size.times(entryPrice.minus(triggerPrice))).div(notionalTotal)
```

Comment:
Might be safer to check `val != 0.0` when doing `.div(val)`

Reply:
ooh good point - I think a valid position should never have 0 notional value, size or leverage - but added to a check above!